From 4c332fdac36f6ed83dffdaa4c9bfbcacb7200141 Mon Sep 17 00:00:00 2001
From: Yang Xiang <yangxian@qti.qualcomm.com>
Date: Tue, 23 Jun 2026 20:04:19 -0700
Subject: [PATCH] Skip Selinux Check

Skip selinux check since we are running in lxc.
And host is android.
Upstream-Status: Inappropriate [embedded specific]
Signed-off-by: yangxian@qti.qualcomm.com
---
 src/core/selinux-access.c | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/src/core/selinux-access.c b/src/core/selinux-access.c
index 62181a6309..4a2394512d 100644
--- a/src/core/selinux-access.c
+++ b/src/core/selinux-access.c
@@ -25,6 +25,7 @@
 #include "strv.h"
 
 static bool initialized = false;
+static bool access_disabled = true;
 
 struct audit_info {
         sd_bus_creds *creds;
@@ -141,29 +142,20 @@ static int access_init(sd_bus_error *error) {
         if (!mac_selinux_use())
                 return 0;
 
-        if (initialized)
+        if (initialized || access_disabled) {
                 return 1;
+        }
 
         if (avc_open(NULL, 0) != 0) {
                 r = -errno;  /* Save original errno for later */
 
                 bool enforce = security_getenforce() != 0;
                 log_full_errno(enforce ? LOG_ERR : LOG_WARNING, r, "Failed to open the SELinux AVC: %m");
-
-                /* If enforcement isn't on, then let's suppress this error, and just don't do any AVC checks.
-                 * The warning we printed is hence all the admin will see. */
-                if (!enforce)
-                        return 0;
-
-                /* Return an access denied error based on the original errno, if we couldn't load the AVC but
-                 * enforcing mode was on, or we couldn't determine whether it is one. */
-                errno = -r;
-                return sd_bus_error_setf(error, SD_BUS_ERROR_ACCESS_DENIED, "Failed to open the SELinux AVC: %m");
+                return 0;
         }
 
         selinux_set_callback(SELINUX_CB_AUDIT, (union selinux_callback) { .func_audit = audit_callback });
         selinux_set_callback(SELINUX_CB_LOG, (union selinux_callback) { .func_log = log_callback });
-
         initialized = true;
         return 1;
 }
@@ -196,9 +188,13 @@ int mac_selinux_access_check_internal(
         assert(error);
 
         r = access_init(error);
+
         if (r <= 0)
                 return r;
 
+        if (access_disabled) {
+                return 0;
+        }
         /* delay call until we checked in `access_init()` if SELinux is actually enabled */
         enforce = mac_selinux_enforcing();
 
-- 
2.34.1

